libkbfs/tlf_journal.go

```diff
@@ -5,22 +5,39 @@
 package libkbfs
 
 import (
+	"errors"
 	"fmt"
 	"path/filepath"
 	"sync"
 
 	"github.com/keybase/client/go/logger"
-	"github.com/keybase/client/go/protocol/keybase1"
 	"golang.org/x/net/context"
 )
 
 // tlfJournalConfig is the subset of the Config interface needed by
 // tlfJournal (for ease of testing).
 type tlfJournalConfig interface {
+	BlockSplitter() BlockSplitter
 	Codec() Codec
 	Crypto() Crypto
-	KBPKI() KBPKI
+	currentInfoGetter() currentInfoGetter
+	encryptionKeyGetter() encryptionKeyGetter
```

Comment:
I tried this, but it makes the adapter annoying, as instead of writing e.g. `func (adapter) currentInfoGetter()`, you'd have to write forwarders for all of cig's methods, or add a KBPKI member to the adapter -- but the latter makes it difficult to switch out the KBPKI goroutine-safely, so that's not great.

FWIW, this whole thing might change once I start trying to carve out some stuff to move out of libkbfs into their own package.